### ISSUE TITLE
much slower backups after upgrade from 1.0.9 to 1.1.8

### ISSUE THREAD

**Issue by Lky** [borgbackup/borg#4246]
## Have you checked borgbackup docs, FAQ, and open Github issues?
Yes

## Is this a BUG / ISSUE report or a QUESTION?
Issue

## System information. For client/server mode post info for both machines.
borg 1.0.9 (Debian Jessie) [Source] / borg 1.0.9 (Debian Stretch) [Repo via SSH)
upgrade to
borg 1.1.8 (Binary) [Source] / borg 1.1.8 (Debian Stretch Backports) [Repo via SSH)

#### Hardware / network configuration, and filesystems used.
ext3 [Source]
SSH
ext4 [Repo]

#### How much data is handled by borg?
Files: 554747 / 175.21 GB

#### Full borg commandline that lead to the problem (leave away excludes and passwords)
borg create -C lz4 --exclude-caches --exclude-from "${BACKUP_EXCLUDE}" -v -s borg@${SERVER}:data::'{hostname}{utcnow:%Y-%m-%d%H%M%S}' /source (3 dirs)

## Describe the problem you're observing.
First run after switch from 1.0.9 to 1.1.8.
Duration: 4 hours 18 minutes 11.14 seconds / Files: 554747 / 175.21 GB
Chunks: 1039359 unique/56809737 total

1.0.9 = Duration: 11 minutes 15.39 seconds / 554742 files
1.1.8 = Duration: 2 hours 29 minutes 17.52 seconds / 554358 files

I had two source jobs, /data (the log above) and /data/group (a subset of the data) and push to the same repo. Subset borg job goes from 2 mins to 25 mins.

maybe related:
https://github.com/borgbackup/borg/issues/2245
https://github.com/borgbackup/borg/issues/3239

**Comment by dragetd** [comment]
Is this still the case on subsequent backups? The segment compacting was changed between those versions, causing borg to repack all segments files. This takes a while on the first run, but should be back to normal after the first run.

**Comment by ThomasWaldmann** [comment]
use `borg create --list --filter=AME ...` to see what borg considers added or modified.
if you see a lot of M and you think they are not modified, check ctime, inode number, size changes.

you could also use strace with some timing options for old and new borg to see why it is slow.

**Comment by Lky** [comment]
Okay i did some tests, it lists some M files which don't really change, but they got a access time update.

I also got the issue solved via https://github.com/borgbackup/borg/issues/3239 --nobsdflags restores nearly full speed.

I did a search about --nobsdflags and now i can see what i missed all the time.

The thing is, is there a comment in the changelog or the readme about this performance drop after upgrade to 1.1, because of this flag? I'm not so sure how this could be more visible.

Sorry for creating this bug report, i should read the older ones and readme maybe with more care.
